### PR TITLE
Added real part and imaginary part methods

### PR DESCRIPTION
**Pull request by YakoYakoYokuYoku**
I've just added this two methods 'cause i require them in my projects.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -143,6 +143,16 @@ impl<T: Clone + Num + Neg<Output = T>> Complex<T> {
...
 impl<T: Clone + Float> Complex<T> {
+    /// Returns the real part.
+    #[inline]
+    pub fn real(&self) -> T {
+        self.re
+    }
+    /// Returns the imaginary part.
+    #[inline]
+    pub fn imag(&self) -> T {
+        self.re
```
Comment (cuviper) [inline src/lib.rs]: That's the wrong field.

Comment (YakoYakoYokuYoku) [reply src/lib.rs]: You are right, fixing this :wrench:

**Comment by YakoYakoYokuYoku** [comment]
> The fields are public -- why do you want method accessors?

Because when I try to do in my projects to return the real or imaginary part (i.e.: [the gamma function](https://rosettacode.org/wiki/Gamma_function#C.23)), I definitely need a real part method for both num-traits or num-complex crates and to follow standards (like the Python default library has got also these methods).

**Comment by cuviper** [comment]
Still -- you can already access any `value.re` or `value.im`, so why should these be methods?

If we *were* to design methods, then it should be like `fn real(&self) -> &T`, borrowing the value instead of only implementing this for `T: Float` (and thus `Copy`). Then we'd also probably want `real_mut` for mutable access. But such methods are strictly less powerful than direct field access.

I can somewhat see a benefit of trait methods for this, which I'll comment about on your num-traits PR. However, the methods you're adding here will be completely independent of that. These are *inherent* methods, which are independent of the implementation of any trait methods. The only way to override a trait method is in the actual `impl Trait for Foo` block.

(Perhaps you are used to Go? That has a structural type system, where an interface is implemented just by having the matching methods. Rust uses a nominal type system, which is more explicit.)

**Comment by YakoYakoYokuYoku** [comment]
Recently I've been making changes (attrocities and chaos :smiling_imp: ) to my forks of num-complex and num-traits for an implementation of a "CommonFloat" trait (as for gaussian integers, it will be another day) and then I will sub-sequentially merge these pull requests in favour to a "CommonFloat", as for know all I can do is show you the [crazy](https://github.com/YakoYakoYokuYoku/num-traits) [prototypes](https://github.com/YakoYakoYokuYoku/num-complex) i needed to do for my [horse-shoe](https://github.com/YakoYakoYokuYoku/HorseShoe) project (a calculus library).

**Comment by YakoYakoYokuYoku** [comment]
Closing in favour of #40.

**Pull request closed without merging**